visualization/app/codeCharta/state/store/fileSettings/edges/edges.merger.ts

```diff
@@ -19,18 +19,18 @@ export function getMergedEdges(inputFiles: CCFile[], withUpdatedPath: boolean):
 					attributes: oldEdge.attributes,
 					visible: oldEdge.visible
 				}
-				const equalEdgeItem = edges.find(e => e.fromNodeName == edge.fromNodeName && e.toNodeName == edge.toNodeName)
+				const equalEdgeItem = edges.get(`${edge.fromNodeName}|${edge.toNodeName}`)
 
-				if (equalEdgeItem) {
+				if (equalEdgeItem !== undefined) {
 					for (const key in edge.attributes) {
 						equalEdgeItem.attributes[key] = edge.attributes[key]
+						edges.set(`${edge.fromNodeName}|${edge.toNodeName}`, equalEdgeItem)
```

Comment:
```suggestion
```

This call is just going to replace itself. `equalEdgeItem` is already a reference and as such, you'll change the actual entry inside the map implicitly.

And even if that would not be the case, it would actually be better to place the set after the for loop so that it's just called once.

As a change independent comment: using `for ... in` should ideally not be used since it'll also iterate over the prototype. Instead, it's better to use e.g., `for (const key of Object.keys(edge.attributes))` or `for (const [key, value] of Object.entries(edge.attributes))`.
And is it actually intentional to override existing keys? I just wonder if that's always the best behavior.

Reply:
I couldn't tell whether or not overriding the values of the keys here was intentional or not. There exists an test to ensure this is the case, so I thought it was important.
The question is: When can two edges have the same source and destination?
- In a single file, this shouldn't be the case.
- In multiple mode, it's impossible, because the paths would be different. We use the fileName as the name of the folder we put as a root for each project and fileNames are unique in codecharta. 
- In delta mode, there might be a lot of these edges, but edges don't make sense in delta mode actually. We can't display a delta of those. Edges should be disabled there in my opinion.